testdata/src/http_request/http_request.go

```diff
@@ -14,108 +14,108 @@ func _() {
 
 	ctx := context.Background()
 
-	req, _ := http.NewRequest(http.MethodPost, url, nil) // want `should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext`
+	req, _ := http.NewRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
 	cli.Do(req)
 
 	req2, _ := http.NewRequestWithContext(ctx, http.MethodPost, url, nil) // OK
 	cli.Do(req2)
 
-	req3, _ := http.NewRequest(http.MethodPost, url, nil) // OK
-	req3 = req3.WithContext(ctx)
+	req3, _ := http.NewRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
+	req3 = req3.WithContext(ctx)                          // want `\(\*net/http\.Request\)\.WithContext must not be called. use net/http\.NewRequestWithContext`
 	cli.Do(req3)
 
 	f2 := func(req *http.Request, ctx context.Context) *http.Request {
 		return req
 	}
-	req4, _ := http.NewRequest(http.MethodPost, url, nil) // want `should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext`
+	req4, _ := http.NewRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
 	req4 = f2(req4, ctx)
 
-	req41, _ := http.NewRequest(http.MethodPost, url, nil) // OK
-	req41 = req41.WithContext(ctx)
+	req41, _ := http.NewRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
+	req41 = req41.WithContext(ctx)                         // want `\(\*net/http\.Request\)\.WithContext must not be called. use net/http\.NewRequestWithContext`
 	req41 = f2(req41, ctx)
 
 	newRequest := http.NewRequest
-	req5, _ := newRequest(http.MethodPost, url, nil) // want `should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext`
+	req5, _ := newRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
 	cli.Do(req5)
 
-	req51, _ := newRequest(http.MethodPost, url, nil) // OK
-	req51 = req51.WithContext(ctx)
+	req51, _ := newRequest(http.MethodPost, url, nil) // want `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
+	req51 = req51.WithContext(ctx)                    // want `\(\*net/http\.Request\)\.WithContext must not be called. use net/http\.NewRequestWithContext`
 	cli.Do(req51)
 
-	req52, _ := newRequestPkg(http.MethodPost, url, nil) // want `should rewrite http.NewRequestWithContext or add \(\*Request\).WithContext`
+	req52, _ := newRequestPkg(http.MethodPost, url, nil) // TODO: false negative `net/http\.NewRequest must not be called. use net/http\.NewRequestWithContext`
```

Comment:
false negative we have to send pull request to https://github.com/gostaticanalysis/analysisutil/